### ISSUE TITLE
No results anymore, how to debug this? No exception, was working fine for months.

### ISSUE THREAD

**Issue by tobiasvielmetter** [pgrimaud/instagram-user-feed#45]
Hi,

 ran into this issue that since today I don't get any results anymore, even though I didn't change anything. I checked and the handle is fine and also there is no exception thrown. Then I updated composer packages just to be sure, but nothing changed.

Is this package still working or did Instagram block it?

I am using the default ^5 example code and current release.

```
$cache = new Instagram\Storage\CacheManager(__DIR__ . '/../../../cache/');
		$api = new Instagram\Api($cache);
		$api->setUserName(get_sub_field( 'instagram_handle' ));
		$instagram_html = "";

		try {
			$feed = $api->getFeed();
			$max_count = get_sub_field( 'amount_of_posts' );
			$i = 0;
			$instagram_html .= '<div class="row">';
			$medias = $feed->getMedias();

			foreach ($medias as $media) {
				$i++;
				$instagram_html .= '<div class="col';
				if($i>3) {
					$instagram_html .= ' d-none d-lg-block';
				} else if($i>2) {
					$instagram_html .= ' d-none d-md-block';
				}
				$instagram_html .= '"><a href="'.$media->getLink().'" target="_blank" title="'.$media->getCaption().'" alt="'.$media->getCaption().'">
				<img class="dshadow" src="/wp-content/themes/understrap-child/img/Rectangle.png" width="100%" style="background-repeat: no-repeat;background-size: cover;background-image:url('.$media->displaySrc.');" /></a></div>';
				if($max_count<=$i) break;
			}
			$instagram_html .= '</div>';
		} catch (Exception $exception) {
			print_r($exception->getMessage());
		} catch (\GuzzleHttp\Exception\GuzzleException $e) {
			print_r($exception->getMessage());
		}
		echo $instagram_html;
```

**Comment by tobiasvielmetter** [comment]
@pgrimaud sorry I accidentally closed the issue, its not fixed though. Could you reopen it?

**Comment by pgrimaud** [comment]
Hi @tobiasvielmetter,

I just tried your script and it worked fine.

![image](https://user-images.githubusercontent.com/1866496/63529993-f48e8500-c505-11e9-8625-4dce5d8ee1bc.png)

Did you try from another IP? Maybe instagram blocked yours.

**Comment by tobiasvielmetter** [comment]
@pgrimaud  ha, thats strange, i will try in a moment on another ip and do a deeper investigation. Its on a shared webhost, so in case the ip is blocked we would be screwed and need another solution... anyhow, I will get back to you once I have more precise info. Thank you!

**Comment by fersk** [comment]
I can confirm the same issue. Looks related with IP, maybe some kind of rate limiting.

**Comment by tobiasvielmetter** [comment]
Ok, I can confirm that it all works from another server. So this is either a IP blocking issue or a environment issue. Although I strongly tend toward IP blocking at the moment, as there is no exception and it was working fine for months.

**Comment by pgrimaud** [comment]
I will try to thrown an exception when Instagram triggers a rate limiting.

**Comment by jakejohns** [comment]
Right now, I believe im experiencing the same issue.
Works on one server, not the other, so likely IP related.

> ErrorException (E_WARNING)
> Invalid argument supplied for foreach()
> Instagram/Hydrator/HtmlHydrator.php:33

**Comment by barryjenkin** [comment]
fetchrss.com posted this message in the last 2 days:
Instagram has just changed the behavior of its web version interface. Now no one can view more than a few users' profiles without being authorized on their servers first. In other words, user profiles are no longer publicly available for servers like ours. 
We're currently searching for ways to solve this problem. Sorry for the inconveniences.

**Comment by pgrimaud** [comment]
Weird...

Are you on shared server for your hosting?

**Comment by UnbekanntesPferd** [comment]
I have the exact same problems since some days.
Using a shared hosting.
Anyone tried if it's the same issue with token usage?

**Comment by asylwan** [comment]
I'm also having issues since a few days ago. It used to work fine but doesn't anymore. I tried the same script from three different servers and get the same issue.

```[27-Aug-2019 13:21:06 UTC] PHP Fatal error:  Uncaught GuzzleHttp\Exception\ClientException: Client error: `GET https://www.instagram.com/graphql/query/?query_hash=42323d64886122307be10013ad2dcc44&variables=%7B%22id%22:%22201012318%22,%22first%22:%2212%22,%22after%22:%7B%7D%7D` resulted in a `400 Bad Request` response:
{"message": "execution failure", "errors": [{"message": "execution error", "locations": []}], "data": {"user": null}, "s (truncated...)```

**Comment by pgrimaud** [comment]
@asylwan Can you provide the code you've written?

**Comment by danielwaghorn** [comment]
I've recently experienced the same issue from shared hosting and have managed to work round it by specifying the IP address to use via the cURL config supplied to Guzzle.

**N.B** This works since I have a dedicated IP address attached to the shared hosting account in this instance, therefore we can access Instagram outwith the shared host's IP address which is likely being blocked.

I use an environment variable to force the outbound IP if required, otherwise we'll just let the package instantiate Guzzle as normal.

```
// Support forcing client IP for shared environments
// Stops Instagram blocking request
$client = NULL;

if (env('GUZZLE_OUTBOUND_IP')) {
   $client = new GuzzleHttp\Client([
      'curl' => [
         CURLOPT_INTERFACE => env('GUZZLE_OUTBOUND_IP'),
      ]
   ]);
}

$cache = new CacheManager(storage_path() . '/instagram');
$api   = new Api($cache, $client);
```

**Comment by asylwan** [comment]
> @asylwan Can you provide the code you've written?

Sure, here it is:

```
function getInstagramContent()
    {
        $cache = new Instagram\Storage\CacheManager(get_template_directory() . '/profet-includes/cache/instagram-feed');
        $api   = new Instagram\Api($cache);
        $api->setUserName('loremipsum');

        $feed = $api->getFeed();

        foreach ($feed->medias as $feed_item) {
            $item = array();
            $item['id'] = $feed_item->id;
            $item['caption'] = $feed_item->caption;
            $item['date'] = $feed_item->date;
            $item['link'] = $feed_item->link;
            $item['image'] = $feed_item->displaySrc;
            createPostFromInstagramItem($item);
        }
    }
```

**Comment by joshtipton28** [comment]
I'm running into the same problem. I removed the Instagram feed from my site for a few days and then added it back. It worked for a few hours and then stopped working again. 

A side note, it doesn't look like the cache is actually doing anything with my site either (nothing populates in my cache directory) so it looks like Instagram is throttling/blocking the traffic.

**Comment by pgrimaud** [comment]
I just tried to forward x-csrf-token https://github.com/pgrimaud/instagram-user-feed/commit/28e124f18671a8f888c3266bafeb5dc815a93a6b, maybe it will fix it.

Some one with this error could try version 5.5.3 ? https://github.com/pgrimaud/instagram-user-feed/releases/tag/5.5.3

Thanks

**Comment by asylwan** [comment]
> I just tried to forward x-csrf-token [28e124f](https://github.com/pgrimaud/instagram-user-feed/commit/28e124f18671a8f888c3266bafeb5dc815a93a6b), maybe it will fix it.
> 
> Some one with this error could try version 5.5.3 ? https://github.com/pgrimaud/instagram-user-feed/releases/tag/5.5.3
> 
> Thanks

I just updated to 5.5.3 and tried it out but I'm still getting the same error unfortunately.

**Comment by UnbekanntesPferd** [comment]
Same here.
Unfortunately still not working after the update to 5.5.3.

To reproduce it, maybe try to request instagram consistently for some days (I'm using a cron every 10 minutes).

**Comment by sympho29** [comment]
Hi,
Not working either after update to 5.5.3.

@danielwaghorn's tip (setting Guzzle outbound IP) worked only for a few hours last friday.

I'm using a cron every half-an-hour.

**Comment by pgrimaud** [comment]
I just released a dirty "beta" version with login process : 

`composer req pgrimaud/instagram-user-feed "5.6-beta"`

Actually, there is no BC, but the code is a little bit dirty...

You can take a look at examples files `getProfileWithLogin.php` and `getMediasWithLogin.php`.
You just have to change login / password data.

Someone can try it? If this release fix issues, I will release a v6.0.0 with better code and probably some little BC.

Please open another issue to talk about it. Thanks.

**Comment by giangvt** [comment]
Hi @pgrimaud , i had a same problem with ProfilePage : 
$newCache->setUserId($data->entry_data->ProfilePage[0]->graphql->user->id); ( file HtmlTransportFeed.php)
Can you help me ? What can i do ?
